src/pyhf/optimize/opt_minuit.py

```diff
@@ -99,11 +102,14 @@ def _minimize(
             fitresult (scipy.optimize.OptimizeResult): the fit result
         """
         maxiter = options.pop('maxiter', self.maxiter)
-        # 0: Fast, user-provided gradient
-        # 1: Default, no user-provided gradient
-        strategy = options.pop(
-            'strategy', self.strategy if self.strategy is not None else not do_grad
-        )
+        # int(not do_grad) results in iminuit.Minuit.strategy of either:
+        #   0: Fast. Does not check a user-provided gradient.
+        #   1: Default. Checks user-provided gradient against numerical gradient.
+        strategy = options.pop("strategy", self.strategy)
+        # Guard against None from either self.strategy defaulting to None or
+        # passing strategy=None as options kwarg
+        if strategy is None:
+            strategy = int(not do_grad)
```

Comment:
As the `iminuit.Minuit.strategy` is an integer, instead of having to have a human determine the integer representation of the inverse of the truthiness of `do_grad` during debugging or inspection do this in code.

Reply:
If you're aiming for maximum readability, I like `strategy = 0 if do_grad else 1` best.